newIDE/app/src/VariablesList/index.js

```diff
@@ -149,22 +150,42 @@ export default class VariablesList extends React.Component<Props, State> {
           variable,
           depth + 1,
           index,
-          parentVariable
+          parentVariable,
+          type
         );
       })
     );
   }
 
+  _getInstanceVariableType = (name: string) => {
+    const { variablesContainer, inheritedVariablesContainer } = this.props;
+    return inheritedVariablesContainer // We check for 3 types of variable states, when editing instance variables
+      ? variablesContainer.has(name) && !inheritedVariablesContainer.has(name)
+        ? 'instance' // -variable that is unique to the instance - we can edit its name/structure
+        : !variablesContainer.has(name) && inheritedVariablesContainer.has(name)
+        ? 'object' // -displaying object variable at its default value, changing it will create an inherited
+        : variablesContainer.has(name) && inheritedVariablesContainer.has(name)
+        ? 'inherited' // -instance variable created when editing an object variable's value, deleting it will re-show
+        : '' // object go in inheritedVariablesContainer, the rest stay in variablesContainer
+      : '';
+  };
+
   _renderVariableAndChildrenRows(
     name: string,
     variable: gdVariable,
     depth: number,
     index: number,
-    parentVariable: ?gdVariable
+    parentVariable: ?gdVariable,
+    parentType: ?string = ''
```

Comment:
parentType => parentOrigin

Reply:
Also prefer initialization a string that is not specified to `null`. This make ultra explicit that this is a different case and that this `parentOrigin` is not specified, so must be computed if needed.